docs/src/get-started.md

```diff
@@ -0,0 +1,191 @@
+
+It provides three options for users
+
+1. A command line version [Julia/downloads](http://julialang.org/downloads/) (**recommended**)
+2. A community maintained IDE [Juno](http://www.junolab.org/)
+3. [JuliaBox.com](https://www.juliabox.com/) - a Jupyter notebook in the browser
+
+For command line version, we recommend that you install a version downloaded from Julia's [official website](http://julialang.org/downloads/), as Turing may not work correctly with Julia provided by other sources (e.g. Turing does not work with Julia installed via apt-get due to missing header files).
+
+Juno also needs the command line version installed. This IDE is recommended for heavy users who require features like debugging, quick documentation check, etc.
+
+JuliaBox provides a pre-installed Jupyter notebook for Julia. You can take a shot at Turing without installing Julia on your machine in few seconds.
+
+### Install Turing.jl
+
+Turing is an officially registered Julia package, so the following should install a stable version of Turing:
+
+```julia
+Pkg.add("Turing")
+
+# Switch to master branch; recommended
+Pkg.checkout("Turing", "master")
+Pkg.build("Turing")
+
+Pkg.test("Turing")
+```
+
+[**Recommended**] If you want to use the latest version of Turing with some experimental features, you can try the following instead:
+
+```julia
+Pkg.update()
+Pkg.clone("Turing")
+Pkg.build("Turing")
+Pkg.test("Turing")
+```
+
+If all tests pass, you're ready to start using Turing.
+
+## Basics
+
+### Introduction
+
+A probabilistic program is Julia code wrapped in a `@model` macro. It can use arbitrary Julia code, but to ensure correctness of inference it should not have external effects or modify global state. Stack-allocated variables are safe, but mutable heap-allocated objects may lead to subtle bugs when using task copying. To help avoid those we provide a Turing-safe datatype `TArray` that can be used to create mutable arrays in Turing programs.
+
+For probabilistic effects, Turing programs should use the `~` notation:
+
+`x ~ distr` where `x` is a symbol and `distr` is a distribution. If `x` is undefined in the model function, inside the probabilistic program, this puts a random variable named `x`, distributed according to `distr`, in the current scope. `distr` can be a value of any type that implements `rand(distr)`, which samples a value from the distribution `distr`. If `x` is defined, this is used for conditioning in a style similar to Anglican (another PPL). Here `x` should be a value that is observed to have been drawn from the distribution `distr`. The likelihood is computed using `logpdf(distr,y)` and should always be positive to ensure correctness of inference algorithms. The observe statements should be arranged so that every possible run traverses all of them in exactly the same order. This is equivalent to demanding that they are not placed inside stochastic control flow.
```

Comment:
```
Here `x` should be a value that is observed to have been drawn from the distribution `distr`.
```
->
```
In this case, `x` is an observed value, assumed to have been drawn from the distribution `distr`.
```
maybe?

Reply:
`should always be positive`: the pdf is positive, but the log-pdf need not be.

Reply:
`observe statements`: you have not defined what an `observe statement` is (although I get what you mean by this). We should be as clear as possible I think.

Reply:
I'd like to clarify how the `observe` statements work. If we have a model like:

```julia
@model simple_choice(x, y) = begin
  p ~ Beta(2, 2)
  z ~ Categorical(1)
  for i = 1:length(x)
    if p > 0.5
      x[i] ~ Normal(0, 1)
      y[i] ~ Normal(0, 1)
    else
      y[i]~ Normal(2, 1)
      x[i] ~ Normal(2, 1)
    end
  end
end
```
Where `x` and `y` are observations, does that mean this should error out because `x` and `y` are in different positions in the control flow?

Reply:
I think it shouldn’t.

Reply:
Okay, strange. The above example compiles and samples, is this some kind of an error in the compiler? Or my fork of the compiler, anyway?